api/src/pcapi/alembic/versions/20231123T163324_d158e4b24ff2_alter_user_nullable_password.py

```diff
@@ -0,0 +1,27 @@
+"""
+Make user password nullable.
+Add not null password or existing single sign-on trigger check.
+"""
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+from pcapi.core.users.models import User
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "d158e4b24ff2"
+down_revision = "197afaebf455"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.alter_column("user", "password", existing_type=postgresql.BYTEA(), nullable=True)
+    User.trig_ensure_password_or_sso_exists_ddl(target=None, bind=op.get_bind())
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏warning ici: si `User.trig_ensure_password_or_sso_exists_ddl` change de nom ou de fichier, la migration va fail.
On peut mitiger en ajoutant un comm dans User
Ou recopier la fonction ici (tant pis pour la duplication)

Je veux bien l'avis de devs plus confirmés là dessus (@dbaty @xordoquy ou @francois-seguin ?)